sql/src/main/java/io/crate/planner/consumer/QueryAndFetchConsumer.java

```diff
@@ -119,7 +122,18 @@ private Plan normalSelect(QueriedTableRelation table, ConsumerContext context) {
              * orderByInputs:       [in(0), in(2)]         // for topN projection on shards/collectPhase AND handler
              * finalOutputs:        [in(0), in(1)]         // for topN output on handler -> changes output to what should be returned.
              */
-            List<Symbol> qsOutputs = querySpec.outputs();
+            List<Symbol> qsOutputs = Lists2.copyAndReplace(
```

Comment:
I think this should be done in a join related component and specific to the whereClauses that cannot be pushed down - maybe in the MSS FetchPushDown or ManyTableConsumer.